### PR TITLE
Chain PEtab v2 experiment periods natively in the JAX simulator

### PR DESCRIPTION
**Pull request by FFroehlich**
Previously, PEtab v2 experiments with more than two periods were
collapsed into SBML events at import time via
ExperimentsToSbmlConverter, for both the sundials and JAX backends.
For JAX, this meant period switches were driven by root-finding on
synthetic indicator parameters baked into the compiled model rather
than by directly chaining simulation calls.

For the JAX backend, skip that conversion entirely and instead run
one ODE integration per experiment period directly in
JAXModel.simulate_condition, carrying state and heaviside/event state
across period boundaries the same way pre-equilibration already hands
off into the main simulation. JAXProblem's measurement bucketing,
parameter mapping, and reinitialisation resolution are generalised
from a hardcoded two-phase (preeq + main) model to arbitrary period
counts. The sundials backend is unaffected.

Along the way, fixes several latent bugs that were only reachable
once JAX stopped seeing SBML-converted (indicator-only) condition
tables: condition tables with multiple simultaneous changes, state
reinitialisation lookups against the (long-format) condition table,
a "preequilibration" substring-matching heuristic that depended on
the converter's naming convention, and a couple of shape bugs in
JAXModel for single-state models and models without observable/noise
parameter overrides.

Co-Authored-By: Claude Sonnet 5 <noreply@anthropic.com>
Claude-Session: https://claude.ai/code/session_0173CHAQAGCsTtnyNibrFjDv

### CODE REVIEW

File: python/sdist/amici/sim/jax/model.py [hunk 1]
```diff
@@ -702,9 +670,161 @@ def simulate_condition_unjitted(
...
+                x_solver,
+                p_i,
+                tcl_i,
+                root_finder,
+                self._root_cond_fn,
+                self._delta_x,
+                h_mask,
+                h,
+                {},
+            )
```
[the remark this answers is not in the document]
Comment (FFroehlich) [reply python/sdist/amici/sim/jax/model.py:919]: This is intentional rather than dead code: `h` here is the *previous* period's ending heaviside/event state, and reusing it via `_handle_t0_event`'s existing `h_preeq`-nonempty short-circuit for `i > 0` is how event state gets carried across period boundaries — the same mechanism already used for the pre-equilibration → main-period handoff. Guarding the call to only run for `i == 0` would be equivalent in behavior but drop that shared code path.

That said, this does mean the event trigger is **not** re-evaluated against the actual state at a period's t0 for `i > 0`. If a period boundary combines a state reinitialisation/parameter change with a model that has genuine events, and the reinit crosses a trigger's threshold, the trigger won't be picked up until (or unless) the ODE integrator crosses the root again during that period. Added `test_event_heaviside_state_not_reevaluated_after_period_reinit` in `python/tests/petab_/test_petab_v2_multiperiod.py` (commit 3e58c65) to pin down and document this current behavior for that edge case, since there wasn't any existing coverage combining multi-period reinitialisation with events. Leaving this thread open rather than resolving it, since it documents a known limitation rather than a fix.

---
_Generated by [Claude Code](https://claude.ai/code)_

[the remark this answers is not in the document]
Comment (FFroehlich) [reply python/sdist/amici/sim/jax/model.py:919]: Update: on reflection this carry-over behavior was actually pathological, not just an untested edge case — a period boundary (or the reinitialisation applied after preequilibration) can cross an event's trigger threshold, and the old behavior would silently keep using the stale pre-transition heaviside state until the ODE integrator happened to cross the root again.

Fixed in 2932169: `_handle_t0_event` now always re-evaluates the trigger condition against the actual incoming state at every period boundary, using the previous heaviside state only as the pre-transition reference for detecting a crossing (exactly as already done for a genuine t=0), instead of short-circuiting. Updated `test_event_heaviside_state_not_reevaluated_after_period_reinit` (renamed to `test_event_heaviside_state_reevaluated_after_period_reinit`) to assert the corrected result. Verified against the full JAX PEtab v2 multiperiod suite, the SBML JAX semantic test suite (including all event-tolerance-flagged cases), and the JAX performance regression suite's event models — all green. Resolving this thread.

---
_Generated by [Claude Code](https://claude.ai/code)_
